src/react-apps/applications/runtime/src/components/base/DatepickerComponent.tsx

```diff
@@ -26,20 +26,28 @@ export class DatepickerComponent
     };
   }
 
-  public onDateChange = () => {
+  public onDateChange = (e: any) => {
+    this.setState({
+      value: e.target.value,
+    });
+  }
+
+  public ondateBlur = () => {
     setTimeout(() => {
-      if (!this.myDateCmp.current.value) {
+      if (!this.myDateCmp.current.value || this.state.value === this.myDateCmp.current.value) {
```

Comment:
This `myDateCmp`, could it be named something that clearly states it's a ref? Like selfRef or datepickerRef?